### PR TITLE
[WIP] Add /consider-arrangement proxy endpoint

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- [ ] Bugfix
- [x] Feature
- [ ] Documentation
- [ ] Other

**Required reviews:** 
- [ ] 1
- [x] 2
- [ ] 3

**What this does:**
- Adds `/consider-arrangement` Porter endpoint in order to enable `nucypher-ts` in the browser.

**Issues fixed/closed:**
- #2763

**Notes for reviewers:**
- This PR is subject to change per discussion in #2763. I'm putting it out here in order to track this issue and reference it in `nucypher-ts` docs - Early `nucypher-ts` users could find it useful.

### CODE REVIEW

File: nucypher/control/controllers.py [hunk 1]
```diff
@@ -310,14 +310,25 @@ def handle_request(self, method_name, control_request, *args, **kwargs) -> Respo
                            self.emitter.MethodNotFound)
 
         try:
+            # handle request headers
+            request_headers = control_request.headers
+            request_headers = {header.upper(): value for (header, value) in request_headers.items()}
```
Comment (KPrasch) [inline nucypher/control/controllers.py:315]: Heh, the headers must begin with an uppercase letter?

Also, just a style comment, parenthesis are not needed here:
```suggestion
            request_headers = {header.upper(): value for header, value in request_headers.items()}
```

File: nucypher/control/controllers.py [hunk 2]
```diff
@@ -310,14 +310,25 @@ def handle_request(self, method_name, control_request, *args, **kwargs) -> Respo
...
+
             request_data = control_request.data
-            request_body = json.loads(request_data) if request_data else dict()
+            # handle bytes in request body
+            if request_headers.get('CONTENT-TYPE') == 'application/octet-stream':
+                request_body = {"data": bytes(request_data)}
```
Comment (KPrasch) [inline nucypher/control/controllers.py:320]: If the request header specifies a content type of bytes, isn't the body already bytes?  Is this just a matter of getting the bytes into a dictionary for consumption?

File: nucypher/control/controllers.py [hunk 3]
```diff
@@ -310,14 +310,25 @@ def handle_request(self, method_name, control_request, *args, **kwargs) -> Respo
...
+            else:
+                request_body = json.loads(request_data) if request_data else dict()
 
             # handle query string parameters
             if hasattr(control_request, 'args'):
                 request_body.update(control_request.args)
 
             request_body.update(kwargs)
+            if request_headers.get('X-PROXY-DESTINATION'):
+                request_body['proxy_destination'] = request_headers['X-PROXY-DESTINATION']
```
Comment (KPrasch) [inline nucypher/control/controllers.py:331]: Hmm it's unclear what is meant here, is `proxy_destination` an alias for `X_PROXY_DESTINATION`?

File: nucypher/utilities/porter/porter.py [hunk 4]
```diff
@@ -160,7 +161,14 @@ def retrieve_cfrags(self,
...
         client = RetrievalClient(self)
         return client.retrieve_cfrags(treasure_map, retrieval_kits,
-            alice_verifying_key, bob_encrypting_key, bob_verifying_key, policy_encrypting_key)
+                                      alice_verifying_key, bob_encrypting_key, bob_verifying_key, policy_encrypting_key)
+
+    def proxy_consider_arrangement(self, ursula_uri: str, data: bytes):
+        response = requests.post(f"{ursula_uri}/consider_arrangement", data=data, verify=False,
+                                 headers={'Content-Type': 'application/octet-stream'})
+        if response.status_code >= 400:
+            raise RuntimeError(response.content)
```
Comment (KPrasch) [inline nucypher/utilities/porter/porter.py:170]: Is `RuntimeError` used to handle other 500s too?

**Comment by derekpierre** [comment]
See comment here - https://github.com/nucypher/nucypher/issues/2763#issuecomment-930489105.

**Review by derekpierre — CHANGES_REQUESTED** [review]
Just formalizing that this PR should be closed based on the result of discussions documented here - https://github.com/nucypher/nucypher/issues/2763#issuecomment-930489105.

**Comment by piotr-roslaniec** [comment]
Closing without merging since these changes are superfluous per discussion in #2763.

**Pull request closed without merging**